### PR TITLE
Note about OData upsert

### PR DESCRIPTION
**Pull request by artemkovalyov**
## What Has Changed?

A note on upserting

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

**Comment by artemkovalyov** [comment]
@newtork , this a promised note about UPSERT that should link to your example of executing it with Generic OData client.

File: docs/java/features/odata/type-safe-client-odata-v4.mdx [hunk 1]
```diff
@@ -94,6 +94,16 @@ Request parameters:
   - `ignoreAnyVersionIdentifier()` will ensure that no ETag is sent.
 - All operations allow for adding custom headers via `withHeader(...)`
 
+:::note  "`UPSERT` operation in OData v4"
+
+We do not provide an API for `UPSERT` [operation available in OData v4](http://docs.oasis-open.org/odata/odata/v4.0/os/part1-protocol/odata-v4.0-os-part1-protocol.html#_Toc372793720).
+`UPSERT` uses `PATCH` or `PUT` HTTP method which allows updating or creating an Entity if it doesn't exist and URL is valid.
+This gives mostly performance benefits for a narrow set of cases where it matters.
+You can implement `UPSERT` by combining `UPDATE` and `CREATE` requests mentioned above together with custom logic required to handle the upserting.
```
Comment (MatKuhr) [inline docs/java/features/odata/type-safe-client-odata-v4.mdx:102]: For that the user needs to know whether or not the entity already exists. I think UPSERT is instead achieved by simply always using `updateXyz`. @newtork correct me if I am wrong here.

File: docs/java/features/odata/type-safe-client-odata-v4.mdx [hunk 2]
```diff
@@ -94,6 +94,16 @@ Request parameters:
...
+If `UPSERT` is mission critical for you application, you can use our Generic OData client to construct an `UPSERT` call yourself by example.
```
Comment (MatKuhr) [inline docs/java/features/odata/type-safe-client-odata-v4.mdx:103]: > by example

where is the example?

**Review by MatKuhr — CHANGES_REQUESTED** [review]

**Comment by artemkovalyov** [comment]
@newtork , do you still plan to provide a workaround? I assume even POST would work as you can choose a method when dealing with a generic client?

@MatKuhr, if you'd like you can help @newtork with that. I'll leave the PR open and link to an example when it's ready.

**Pull request closed without merging**